scipy/sparse/_compressed.py

```diff
@@ -124,14 +125,42 @@ def _getnnz(self, axis=None):
             axis, _ = self._swap((axis, 1 - axis))
             _, N = self._swap(self.shape)
             if axis == 0:
-                return np.bincount(downcast_intp_index(self.indices),
-                                   minlength=N)
+                return np.bincount(downcast_intp_index(self.indices), minlength=N)
             elif axis == 1:
                 return np.diff(self.indptr)
             raise ValueError('axis out of bounds')
 
     _getnnz.__doc__ = _spbase._getnnz.__doc__
 
+    def count_nonzero(self, axis=None):
+        self.sum_duplicates()
+        if axis is None:
+            return np.count_nonzero(self.data)
+
+        if self.ndim == 1:
+            if axis not in (0, -1):
+                raise ValueError('axis out of bounds')
+            return np.count_nonzero(self.data)
+
+        if axis < 0:
+            axis += 2
+        axis, _ = self._swap((axis, 1 - axis))
+        if axis == 0:
+            _, N = self._swap(self.shape)
+            mask = self.data != 0
+            idx = self.indices if mask.all() else self.indices[mask]
+            return np.bincount(downcast_intp_index(idx), minlength=N)
+        elif axis == 1:
+            if self.data.all():
+                return np.diff(self.indptr)
+            x, y = itertools.tee(self.indptr)
```

Comment:
https://docs.python.org/3/library/itertools.html#itertools.pairwise

Reply:
The python docs say `itertools.pairwise is only for Python 3.10 and later (we use it a lot in networkx, though we built our own long ago).  Scipy supports 3.9 still. (will that change soon?)

We could create something in  sputils.py  and drop it after 3.9 is dropped. But that would be more trouble than this code (I think).

Reply:
Ahh!  I read the page wrong -- Scipy supports 3.10+ now. So `itertools.pairwise` will work!  Thanks!